### DISCUSSION TITLE
How can I record video?

### DISCUSSION THREAD

**Discussion by des-zhong** [Toni-SM/skrl#166] [Q&A]
I'm runing isaacgymEnvs on a remote server. How can I visualize the process?
I saw a capture_video option in function load_isaacgym_env_preview3 but followed by "#to do". Does it means it's not implemented yet?

**Comment by Toni-SM** [comment]
Hi @des-zhong 

For Isaac Gym preview, the `config.capture_video` just enable the [creation](https://github.com/isaac-sim/IsaacGymEnvs/blob/aeed298638a1f7b5421b38f5f3cc2d1079b6d9c3/isaacgymenvs/tasks/base/vec_task.py#L226) of a [pyvirtualdisplay's SmartDisplay](https://pypi.org/project/PyVirtualDisplay/) used in the Isaac Gym [render](https://github.com/isaac-sim/IsaacGymEnvs/blob/aeed298638a1f7b5421b38f5f3cc2d1079b6d9c3/isaacgymenvs/tasks/base/vec_task.py#L510) method to return a RGB image. Then by wrapping the loaded environment using the [gym.wrappers.RecordVideo](https://www.gymlibrary.dev/api/wrappers/#available-wrappers) wrapper, before wrapping it for skrl, you should be able to record a video.

Note that there is a way to livestream NVIDIA Isaac Gym preview from a remote server. 
If you want to [stream from a remote server](https://isaac-sim.github.io/IsaacLab/source/deployment/docker.html#webrtc-streaming), you will need to migrate to NVIDIA [Isaac Lab](https://isaac-sim.github.io/IsaacLab/index.html).
